src/common/mfu_io.c

```diff
@@ -2080,6 +2171,78 @@ int mfu_rmdir(const char* dir)
     return rc;
 }
 
+int daos_rmdir(const char* dir, mfu_file_t* mfu_file)
+{
+#ifdef DAOS_SUPPORT
+    char* name     = NULL;
+    char* dir_name = NULL;
+    parse_filename(dir, &name, &dir_name);
+    assert(dir_name);
+
+    int rc = 0;
+
+    /* Can't delete root */
+    if (name == NULL) {
+        errno = EINVAL;
+        rc = -1;
+        goto out;
+    }
+
+    /* Lookup the parent directory */
+    dfs_obj_t* parent = daos_hash_lookup(dir_name, mfu_file);
+    if (parent == NULL) {
+        rc = -1;
+        goto out;
+    }
```

Comment:
```suggestion
    /* Lookup the parent directory */
    dfs_obj_t* parent = daos_hash_lookup(dir_name, mfu_file);
    if (parent == NULL) {
        // errno is set by daos_hash_lookup, right?  If so, let's add a comment here, since I'll keep forgetting
        rc = -1;
        goto out;
    }
```

Reply:
@adammoody Yes, daos_hash_lookup does set errno. But... most of the daos_ functions call daos_hash_lookup, so I would need to add a similar comment to each function (~20 of them). We are actually working on moving these daos_ IO functions internal to DAOS, in which case they will RETURN an errno code instead of setting errno, so it will be more more explicit.
If it's okay with you, I'd like to just let this be for now.

Reply:
Sure.  Sounds good.